pinc/Project.inc

```diff
@@ -1068,37 +1069,80 @@ function does_project_page_table_exist($projectid)
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
-// Checks if $value has the form of a valid project ID, either in its
-// long form (projectID####) or just the project number (####).
-// If it does, it returns it, otherwise throws an exception.
-function validate_projectID($param_name, $value, $allownull = false)
+// Confirm that $projectid is a well-formed project ID or raise an exception.
+function validate_projectID($projectid)
 {
-    if (!isset($value) && $allownull)
-        return null;
-    if (1 == preg_match('/^projectID[0-9a-f]{13}$/', $value))
-        return $value;
-    if (1 == preg_match('/^[0-9a-f]{13}$/', $value))
-        return "projectID$value";
-    throw new InvalidProjectIDException(sprintf(
-        _("The value of parameter '%1\$s' ('%2\$s') is not a valid projectID."),
-        $param_name,
-        $value
-    ));
+    if (1 != preg_match('/^projectID[0-9a-f]{13}$/', $projectid))
+    {
+        throw new InvalidProjectIDException(sprintf(
+            _("%s is not a valid project ID."),
+            $projectid
+        ));
+    }
+}
+
+// Get a projectid from an array (usually $_GET or $_POST) and validate that
+// it is well-formed. This function mirrors the other get_*() functions in
+// misc.inc.
+function get_projectID_param( $arr, $key, $allownull = false )
+{
+    $projectid = @$arr[$key];
+    if($projectid == NULL && $allownull)
+    {
+        return NULL;
+    }
+
+    try
+    {
+        validate_projectID($projectid);
+    }
+    catch(InvalidProjectIDException $exception)
+    {
+        throw new InvalidProjectIDException(sprintf(
+            _("The value of parameter '%1\$s' ('%2\$s') is not a valid projectID."),
+            $key,
+            $projectid
+        ));
+    }
+    return $projectid;
 }
 
-// Checks if $value contains only legitimate characters for
-// the filename of a page image file in a project.
-// If it does, it returns it, otherwise dies with a warning.
-function validate_page_image_filename($param_name, $value, $allownull = false)
+// Confirm that $page_name is a well-formed page filename or raise an exception.
+function validate_page_image($page_name)
```

Comment:
do we have unit tests for this? It seems simple enough to write one.

Reply:
No we don't but we should - I will fix that.

Reply:
Tests added.

Reply:
Looks great thanks!